Dynatrace/dynatrace-operator
# changes environment and argument ordering to ensure customer settings take highest precedence

## Description

change the ordering of environment and argument processing in statefulset and ActiveGate to always have customer arguments and env vars take highest precedence.

## How can this be tested?

ActiveGate:
- override any of the following arguments should be reflected in the OneAgent: tenant, server, network-zone, operator-version, proxy

statefulset:
- set env vars and verify: DT_GROUP, DT_NETWORK_ZONE, DT_HTTP_PORT

## Checklist
